packages/omgidl-parser/src/IDLNodeProcessor.ts

```diff
@@ -241,47 +253,45 @@ export class IDLNodeProcessor {
     }
   }
 
-  /** Convert to Message Definitions for serialization and usage in foxglove studio's Raw Message panel. Returned in order of original definitions*/
   toMessageDefinitions(): MessageDefinition[] {
-    const messageDefinitions: MessageDefinition[] = [];
+    const idlMsgDefs = this.toAnnotatedMessageDefinitions();
+
+    return idlMsgDefs.map(toMessageDefinition);
+  }
+
+  /** Convert to Message Definitions for serialization and usage in foxglove studio's Raw Message panel. Returned in order of original definitions*/
+  toAnnotatedMessageDefinitions(): AnnotatedMessageDefinition[] {
+    const messageDefinitions: AnnotatedMessageDefinition[] = [];
     const topLevelConstantDefinitions: MessageDefinitionField[] = [];
 
     // flatten for output to message definition
     // Because the map entries are in original insertion order, they should reflect the original order of the definitions
     // This is important for ros2idl compatibility
     for (const [namespacedName, node] of this.map.entries()) {
-      if (node.declarator === "struct") {
-        messageDefinitions.push({
-          name: namespacedName,
-          definitions: node.definitions
-            .map((def) =>
-              this.idlNodeToMessageDefinitionField(toScopedIdentifier([namespacedName, def.name])),
-            )
-            .filter(Boolean) as MessageDefinitionField[],
-        });
-      } else if (node.declarator === "module") {
-        const fieldDefinitions = node.definitions
+      if (
+        node.declarator === "struct" ||
+        node.declarator === "module" ||
+        node.declarator === "enum"
+      ) {
+        const isEnum = node.declarator === "enum";
+        const members = isEnum ? node.enumerators : node.definitions;
+        const definitionFields = members
           .map((def) =>
```

Comment:
Not something for this PR and likely not a concern for the amount of data involved here, but one easy performance win is to avoid making closures inside of tight loops. Maybe harder to avoid here cause you are using a value outside of the closure but something to keep an eye out for.